### PR TITLE
Improve settings

### PR DESCRIPTION
**Pull request by moarychan**
## Changes:
- Fix permission for the actuator endpoint. The API URL of the Spring Boot Actuator has no relation to the context path of the admin UI.
- Add basic info for default /actuator/info.
- Improve logout logic.
- Discovery of the built-in eureka server component.
- Auto-register admin-server.
- Fix Eureka client status down.
 
### Issue description

When used in conjunction with PetClinic microservice apps in the consumption plan service, the app status of all the apps is down.
image

### Root cause
In the underlying k8s, the hostname of each user's app instance is read as an internal domain name, not an IP. The root cause is that the health-check-url is not correct, see below:
image

### Woorkround
Since the secure internal domain is used, it has to use the HTTPS, and use secure port 443, not the 8080.

- Update each customer app configuration, enable switch `eureka.instance.secure-port-enabled=true`
- Update each customer app auto-configuration, add the below beans to override the default beans provided by Eureka client
   ```java
   @Bean
    public ManagementMetadataProvider serviceManagementMetadataProvider() {
        return new AsaDefaultManagementMetadataProvider();
    }

    @Bean
    public EurekaInstanceConfigBean eurekaInstanceConfigBean(InetUtils inetUtils,
                                                             ManagementMetadataProvider managementMetadataProvider) {
    .......
    }
    ```

## Blocking issues
- [ ] SBA only shows 1 instance for each app even the app has 2 instances
- [ ] SBA login failed when the instance count increases to 2

## Others
See more for [Spring Boot 3 support](https://github.com/moarychan/asa-admin/tree/moary/sb3).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,27 +8,50 @@ This repository maintains a customized Spring Boot Admin Server - **Azure Spring
...
-To deploy Spring Boot Admin to Azure Spring Apps using the Azure Spring Apps button, you need below prerequisites:
+Azure Spring Apps Admin provides two usage methods. 
+By default, the Spring Boot Admin application is not protected, 
+and Microsoft Entra ID protection can also be enabled by switching `spring.cloud.azure.admin.entra-id.enabled`.
+
+## Deploy unprotected components to Azure Spring Apps
+
+To deploy an unprotected component, you only need an Azure subscription, and then you can run this Azure Spring Apps Admin with one click button:
+
+<a href="https://yonghui-apps-dev-nubesgen.azuremicroservices.io/deploy.html?url=https://github.com/fangjian0423/asa-admin" data-linktype="external">
```
Comment (fangjian0423) [inline README.md]: https://aka.ms/spring/asa-button?url=https://github.com/fangjian0423/asa-admin

File: README.md [hunk 2]
```diff
@@ -8,27 +8,50 @@ This repository maintains a customized Spring Boot Admin Server - **Azure Spring
...
+- Microsoft Entra ID switch to enable Azure Spring Apps Admin Microsoft Entra ID - Adding `SPRING_CLOUD_AZURE_ADMIN_ENTRAID_ENABLED` environment variable
+- A Microsoft Entra ID administrative roles user
+- Microsoft Entra ID registration application client id to enable Microsoft Entra ID - Adding `AAD_CLIENT_ID` environment variable
+- Microsoft Entra ID registration application client secret to enable Microsoft Entra ID - Adding `AAD_CLIENT_SECRET` environment variable
+- Microsoft Entra ID registration application tenant id to enable Microsoft Entra ID - Adding `AAD_TENANT_ID` environment variable
 
-Then, you can run this Azure Spring Apps Admin by one click button:
+Then, you can run this Azure Spring Apps Admin with one click button:
 
 <a href="https://yonghui-apps-dev-nubesgen.azuremicroservices.io/deploy.html?url=https://github.com/fangjian0423/asa-admin" data-linktype="external">
```
Comment (fangjian0423) [inline README.md]: https://aka.ms/spring/asa-button?url=https://github.com/fangjian0423/asa-admin

File: README.md [hunk 3]
```diff
@@ -8,27 +8,50 @@ This repository maintains a customized Spring Boot Admin Server - **Azure Spring
 
 # What's the benefit of Azure Spring Apps Admin
```
Comment (fangjian0423) [inline README.md]: The repo will move to https://github.com/Azure-Samples/asa-components-spring-boot-admin.

Title update to "What's the benefit of ASA components Spring Boot Admin" looks better.

File: README.md [hunk 4]
```diff
@@ -8,27 +8,50 @@ This repository maintains a customized Spring Boot Admin Server - **Azure Spring
...
-2. Easily deploy with one click Azure Spring Apps button instead of multiple Azure commands
-3. Fully compatible with community Spring Boot Admin
+1. Support integration with Microsoft Entra ID (Original name: Azure AD) to secure your dashboard.
+2. Support on-click Azure Spring Apps button for easy deployment without using multiple Azure commands.
+3. Fully compatible with community Spring Boot Admin.
 
 # Run Azure Spring Apps Admin on Azure Spring Apps
 
-To deploy Spring Boot Admin to Azure Spring Apps using the Azure Spring Apps button, you need below prerequisites:
+Azure Spring Apps Admin provides two usage methods.
```
Comment (fangjian0423) [inline README.md]: ASA components Spring Boot Admin

File: README.md [hunk 5]
```diff
@@ -8,27 +8,50 @@ This repository maintains a customized Spring Boot Admin Server - **Azure Spring
...
+3. Fully compatible with community Spring Boot Admin.
 
 # Run Azure Spring Apps Admin on Azure Spring Apps
 
-To deploy Spring Boot Admin to Azure Spring Apps using the Azure Spring Apps button, you need below prerequisites:
+Azure Spring Apps Admin provides two usage methods. 
+By default, the Spring Boot Admin application is not protected, 
+and Microsoft Entra ID protection can also be enabled by switching `spring.cloud.azure.admin.entra-id.enabled`.
+
+## Deploy unprotected components to Azure Spring Apps
```
Comment (fangjian0423) [inline README.md]: Deploy unprotected Spring Boot Admin to Azure Spring Apps

File: README.md [hunk 6]
```diff
@@ -8,27 +8,50 @@ This repository maintains a customized Spring Boot Admin Server - **Azure Spring
...
+
+To deploy an unprotected component, you only need an Azure subscription, and then you can run this Azure Spring Apps Admin with one click button:
+
+<a href="https://yonghui-apps-dev-nubesgen.azuremicroservices.io/deploy.html?url=https://github.com/fangjian0423/asa-admin" data-linktype="external">
+<img src="https://user-images.githubusercontent.com/58474919/236122963-8c0857bb-3822-4485-892a-445fa33f1612.png" alt="Deploy to Azure Spring Apps" width="200px" data-linktype="relative-path">
+</a>
+
+Once the deployment is complete, you will redirect the application endpoint, then you can access the Spring Boot Admin dashboard.
+
+## Deploy the component integrated with Microsoft Entra ID to Azure Spring Apps
```
Comment (fangjian0423) [inline README.md]: Deploy the Spring Boot Admin integrated with Microsoft Entra ID to Azure Spring Apps

File: README.md [hunk 7]
```diff
@@ -8,27 +8,50 @@ This repository maintains a customized Spring Boot Admin Server - **Azure Spring
...
-- Azure Service Principal client secret to enable Azure Active Directory - Adding AAD_CLIENT_SECRET environment variable
-- Azure Service Principal tenant id to enable Azure Active Directory - Adding AAD_TENANT_ID environment variable
+- Microsoft Entra ID switch to enable Azure Spring Apps Admin Microsoft Entra ID - Adding `SPRING_CLOUD_AZURE_ADMIN_ENTRAID_ENABLED` environment variable
+- A Microsoft Entra ID administrative roles user
+- Microsoft Entra ID registration application client id to enable Microsoft Entra ID - Adding `AAD_CLIENT_ID` environment variable
+- Microsoft Entra ID registration application client secret to enable Microsoft Entra ID - Adding `AAD_CLIENT_SECRET` environment variable
+- Microsoft Entra ID registration application tenant id to enable Microsoft Entra ID - Adding `AAD_TENANT_ID` environment variable
 
-Then, you can run this Azure Spring Apps Admin by one click button:
+Then, you can run this Azure Spring Apps Admin with one click button:
```
Comment (fangjian0423) [inline README.md]: Then, you can run this Spring Boot Admin with one click button

File: src/main/java/com/azure/spring/asa/component/admin/autoconfigure/client/AsaEurekaClientConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.util.StringUtils;
+
+import java.util.Map;
+
+import static org.springframework.cloud.commons.util.IdUtils.getDefaultInstanceId;
+
+/**
+ * Enhanced implementation of {@link EurekaClientAutoConfiguration}, it's compatible with Azure Spring Apps.
```
Comment (fangjian0423) [inline src/main/java/com/azure/spring/asa/component/admin/autoconfigure/client/AsaEurekaClientConfiguration.java]: it's compatible with Azure Spring Apps Standard consumption and dedicated.

**Comment by moarychan** [comment]
This POC canceled, closing it now.

**Pull request closed without merging**